### PR TITLE
Changed response.headers to response.getHeaders() - there is no 'headers' member in EW HttpResponse.

### PR DESCRIPTION
**Pull request by shigeyf**
Changed response.headers to response.getHeaders() 
There is no 'headers' member in EW HttpResponse, so the codes should use response.getHeaders() in order to include all origin response headers to the Edge response.

### CODE REVIEW

**Comment by bmatthew** [comment]
Hey Shige, there may be some confusion on what's going on with these examples.

In each of them the event being consumed is `responseProvider` is being used. For the `responseProvider`, the EdgeWorker only gets a request object and is expected to construct a full response to the client. In the examples you have changed here an intermediate JS object is created to store the to-be-created response contents. For example in edgekv hello world main.js:

```js
    // We choose to always send back a 200 OK with a default greeting
    // and just log any errors in the 'X-EKV-ERROR' response header
    let response = {status: 200, 
                    headers: 
                      {'Content-Type': ['text/html'], 
                       'Content-Language': [content_lang],
                       // Safely Encode the error message to remove unsafe chars
                       // but also replace some encoded strings with safe chars for readability
                       'X-EKV-ERROR': [encodeURI(err_msg).replace(/(%20|%0A|%7B|%22|%7D)/g, " ")]
                      },
                    body: html_body};
    
    // Send Response
    return createResponse(response.status,
                          response.headers,
                          response.body);
```

The `response` consumed in the call to `createResponse` is not the same as the EdgeWorkers API response object, and it's perfectly valid to have a `response.headers` in this context.

So I don't think this change is valid, so I am going to close the PR.

However, if you found it confusing I believe others would as well. It may be beneficial to rename the response variable name or add some additional comments to note this. If you'd like to make a change like that you could open a new PR with that change.

**Comment by shigeyf** [comment]
@bmatthew  Thanks for the confirmation.

In this PR, I would like to focus on the scenario of the following code.

```JavaScript
    return httpRequest(`${request.scheme}://${request.host}${request.path}?${params.toString()}`, options).then((response) => {
        return createResponse(
            response.status,
            response.headers,
            response.body.pipeThrough(XXX)
        );
    });
```

Because there is no member of "headers" in HttpResponse object definition, which is returned by httpRequest() API. Please see the Type definition below:
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/akamai-edgeworkers/index.d.ts#L595

**Type checking with TypeScript detects the error!**
So, I would recommend to change the code to avoid any confusion.


But, I also confirmed my change with getHeaders() was harmful, since all the response headers from the HttpResponse object from httpRequest() were applied to the response from createResponse().
In short, for example, "Content-Length" response header is used with the way of the change.

I have observed that EW does not override the content length header with the right length of the response.body.
Especially in Stream scenario, there is no way in the responseProvider to determine the length of the returned content.
The header should be constructed in an appropriate way.

Maybe it would be good with the following example:

```JavaScript
    return httpRequest(`${request.scheme}://${request.host}${request.path}?${params.toString()}`, options).then((response) => {
        return createResponse(
            response.status,
            {},
            response.body.pipeThrough(XXX)
        );
    });
```

How do you think about it?
Thanks in advance.
-Shige

File: edgeworkers/examples/respond-from-edgeworkers/responseprovider/trace-headers/main.js [hunk 1]
```diff
@@ -93,7 +93,7 @@ export function responseProvider (request) {
 		if (returnInBody(request)){
 			return createResponse(
 				response.status,
-				response.headers,
+				{},
```
[the remark this answers is not in the document]
Comment (shigeyf) [reply edgeworkers/examples/respond-from-edgeworkers/responseprovider/trace-headers/main.js]: @bmatthew Thanks for the suggestion. Yes, it should be good.

**Review by bmatthew — APPROVED** [review]

**Comment by sscian** [comment]
@shigeyf, Seems that some of the changes your requesting where updated and merged. 

I'm closing this PR, and will update the examples in separate changes were necessary based on your suggestions.
